Mage/src/main/java/mage/abilities/effects/ContinuousEffectImpl.java

```diff
@@ -298,6 +307,7 @@ public boolean isInactive(Ability source, Game game) {
             case UntilYourNextTurn:
             case UntilEndOfYourNextTurn:
             case UntilYourNextEndStep:
+            case UntilTheNextEndStep:
```

Comment:
Must use cleanup step, that place for “your” durations only:

![IMG_0867](https://github.com/user-attachments/assets/aa1c9b76-df34-4da7-a35c-f5b32f375072)

Reply:
Cleanup step is too late, it ends 'beginning of next end step' and not cleanup step.

Reply:
Updated comment to reflect that this section is now being used for things other than 'your turn'.

Reply:
That's wrong usage. Don't use conditional durations here. It's a static duration and must be clean due rules -- at the begin of end step like other static duration does. See example with end of turn and clean up:
![shot_240930_222543](https://github.com/user-attachments/assets/c716f3ab-3ccd-4fa9-a315-59e20877b4e7)

Implementation like that:
![shot_240930_222403](https://github.com/user-attachments/assets/c2c6a017-9501-45a2-83f5-ef82d778e838)

Reply:
Done.